products/idn/api/patch-requests.md

```diff
@@ -0,0 +1,500 @@
+    "name": "Ubuntu",
+    "created": "2021-01-22T14:48:58.072Z",
+    "modified": "2023-06-30T13:39:07.456Z"
+}
+
+```
+
+</details>
+
+## PATCH request structure 
+
+A PATCH request involves sending a JSON PATCH document that represents an array of objects. Each object represents a single operation to be applied to the target resource. 
+
+PATCH requests can be simple, or they can be very complex, but they all share the same essential structure. A PATCH request must include an object that specifies exactly one operation to apply to update the resource, as well as exactly one path that represents the target location where the operation is applied. 
+
+This example request has the basic PATCH structure: 
+
+```text
+PATCH https://{tenant}.api.identitynow.com/v3/sources/:id
+```
+
+```json
+[
+  {
+    "op": "replace",
+    "path": "/description",
+    "value": "new description"
+  }
+]
+
+```
+
+This example request uses a "replace" operation to replace the source's existing description with a new value, "new description". This example shows the parts involved in sending a PATCH request. You must specify an operation to apply to the target resource, a path to apply the operation to, and the change you want to make, often in the form of a value or a "from" location for "copy" and "move" operations. 
+
+You can find this example in the [Patch Source](https://developer.sailpoint.com/idn/api/v3/update-source) specification. The API specifications have examples on the right side of the page that you can copy and use to get started. You can tab between the different examples to see a variety of pre-built requests you can use. 
+
+:::note
+
+The ordering of members in JSON objects doesn't affect the response. These examples all specify the operation first, but you would get the same response if you specified the members in any other order too. 
+
+:::
+
+## Specify an operation 
+
+Once you know the ID of the resource you want to update and you have the resource's details, you can start writing your PATCH request. 
+
+The first step is to specify an operation to apply to the target resource. 
+
+Operation objects must have exactly one "op" member, whose value indicates the operation to perform. 
+
+These are the available PATCH operations: 
+
+| Operation | Description | Example |
+| --- | --- | --- |
+| add | Adds a value to the target location. For more information about the "add" operation and how it behaves in different scenarios, refer [here](https://datatracker.ietf.org/doc/html/rfc6902#section-4.1). | This example uses the [Patch Source Schema](https://developer.sailpoint.com/idn/api/v3/update-source-schema) endpoint to add a new "office" attribute to the end of a source schema's array of attributes: ```[{"op": "add", "path": "/attributes/-", "value": {"name": "office", "type": "STRING", "schema": null, "description": "Office Location", "isMulti": false, "isEntitlement": false, "isGroup": false}}]``` |
```

Comment:
In links, we (at least the product docs team) try to use the title of the page. So in this line, the first link could say, "...refer to the [JSON documentation](link) about the add function." rather than "refer [here](link)."

Reply:
Oh yeah, that is better. I'll use that.